Some ARM cpus does not report the cache size or say it is -1

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=866306

Upstream-status: Pending

Index: haveged-1.9.2/src/havegetune.c
===================================================================
--- haveged-1.9.2.orig/src/havegetune.c
+++ haveged-1.9.2/src/havegetune.c
@@ -795,6 +795,9 @@ static int vfs_configInfoCache(
          ctype = vfs_configFile(pAnchor, path, vfs_configType);
          strcpy(path+plen, "size");
          size  = vfs_configFile(pAnchor, path, vfs_configInt);
+         if (size == -1) {
+             size = ctype == 'I' ? GENERIC_ICACHE : GENERIC_DCACHE;
+         }
          cfg_cacheAdd(pAnchor, SRC_VFS_INDEX,  pArgs[1], level, ctype, size);
          }
      }
